src/ipips/ipip-0412.md

```diff
@@ -0,0 +1,260 @@
+occuring multiple times in the requested DAG) will be present in the CAR
+response. Useful when a deterministic block order is used.
+
+It accepts two values:
+- `y`: Duplicate blocks MUST be sent every time they occur during the DAG walk.
+- `n`: Duplicate blocks MUST be sent only once.
+
+When set to `y`, light clients are able to  discard blocks after
+reading them, removing the need for caching in-memory or on-disk.
+
+Setting to `n` allows for more efficient data transfer of certain types of data,
+but introduces additional resource cost on the receiving end.
+
+If the `dups` parameter is not present in the `Content-Type` header, the
+behavior is unspecified, and the CAR response includes an arbitrary list of
+blocks. In this case, the client should assume `n` as the default, but ignore
+duplicates if they are present.
+
+:::warning
+
+The specified parameter does not apply to virtual blocks identified by identity
+CIDs. CAR responses MUST never include these virtual blocks. The parameter in
+question is meant to control the behavior of non-virtual blocks in the
+response. Therefore, it does not have any effect on virtual blocks, and they
+should never be included in the CAR response, no matter if present, or what
+value is set.
+
+:::
+
+### Signaling in Response
+
+The Trustless Gateway MUST always respond with a `Content-Type` header that includes
+information about all supported/known parameters, even if the client did not
+specify them in the request.
+
+The `Content-Type` header format is as follows:
+
+```
+Content-Type: application/vnd.ipld.car;version=1;order=dfs;dups=y
+```
+
+
+Gateway implementations are free to decide on the implicit default ordering or
+other parameters, and use it in responses when client did not explicitly
+specify any matching preference.
+
+Implementations MAY choose to implement only some of the parameters and return
+HTTP 406 Not Acceptable when client requested a response with unsupported one.
+
+## Design rationale
+
+The proposed specification change aims to address the limitations of the
+existing Trustless Gateway specification by introducing a mechanism for
+negotiating the block order in CAR responses.
+
+By allowing clients to indicate their preferred block order, Trustless Gateways
+can cache CAR responses for popular content, resulting in improved performance
+and reduced network load. Clients benefit from more efficient data handling by
+deserializing blocks as they arrive,
+
+We reuse exiting HTTP content type negotiation, and the CAR content type, which
+already had the optional `version` parameter.
+
+### User benefit
+
+The proposed specification change brings several benefits to end users:
+
+1. Improved Performance: Gateways can decide on their implicit default ordering
+   and cache CAR responses for popular content. In turn, clients can benefit
+   from strong `Etag` in ordered (deterministic) responses. This reduces the
```

Comment:
Is a strong `Etag` something we can (and/or want to) support here? There are still some areas that are unspecified even for responses where the blocks and their order are consistent (e.g. `order=dfs&dups=y`).

See https://github.com/ipfs/specs/pull/402#discussion_r1174223244.

The main ones remaining are:
1. CAR version (we can make v1 part of the etag, but then it'll need to change if people try to change the CARv1 format to support EOF without changing the name of the format)
2. What goes in the header of the CARv1?
   - Maybe I've just missed this, is it just the root CID (i.e. before any pathing occurs)?